streetcrm/static/js/inline_ajax.js

```diff
@@ -28,6 +28,27 @@ Sounds confusing?  It's not too hard to use.  Take a look:
   var my_url = url_formatter({"<froob_id>": 35});
 */
 
+/* Extends the autocomplete widget to add an indication that the user should
+   scroll to see more results if a longer list, and to not treat this message
+   as another result option. */
+$.widget( "ui.autocomplete", $.ui.autocomplete, {
+  options: {items: "> *:not(.scroll-indication)"},
```

Comment:
I want to understand this line and the similar one below.  I see the new class which italicizes the help message (thanks!).  What does the `> *:not(` part do?

Reply:
So this is the part that got kind of hairy, but the jQuery UI menu component [has a selector for elements that are parts of the menu that is `"> *"` by default in `options.items`](https://github.com/OpenTechStrategies/streetcrm/blob/master/streetcrm/static/js/jquery-ui-1.11.4.custom/jquery-ui.js#L1386). UI autocomplete inherits from that, and it uses that selector to determine which elements to make options for the autocomplete input. 

Without changing that string, every `li` element in the autocomplete `ul` was turned into an autocomplete option, including the message "Scroll for more results". Adding the `:not(.scroll-indication)` ignores that element so that the message is still visible without becoming a select option. Overriding it on the UI menu prototype was necessary (I think) because the function for taking internal elements and making them into options is in the UI menu and wasn't being changed by just specifying `options.items` on the altered autocomplete widget. If there's a better way of handling that let me know!

Reply:
That's fascinating!  Thank you for the reference.  I don't know of a better way to do it, so I think we can go with this.